Ray.jl/src/object_store.jl

```diff
@@ -26,13 +16,24 @@ if run in an `@async` task.
 If the task that generated the `ObjectID` failed with a Julia exception, the
 captured exception will be thrown on `get`.
 """
-get(oid::ray_jll.ObjectIDAllocated) = _get(take!(ray_jll.get(oid)))
-get(obj::SharedPtr{ray_jll.RayObject}) = _get(take!(ray_jll.GetData(obj[])))
+get(oid::ray_jll.ObjectIDAllocated) = _get(ray_jll.get(oid))
+get(obj::SharedPtr{ray_jll.RayObject}) = _get(ray_jll.GetData(obj[]))
 get(x) = x
 
-function _get(data::Vector{UInt8})
-    result = deserialize(IOBuffer(data))
+function _get(buffer)
+    result = from_serialized_buffer(buffer)
     # TODO: add an option to not rethrow
     # https://github.com/beacon-biosignals/ray_core_worker_julia_jll.jl/issues/58
     result isa RayRemoteException ? throw(result) : return result
 end
+
+function to_serialized_buffer(data)
+    bytes = Vector{UInt8}()
+    io = IOBuffer(bytes; write=true)
+    serialize(io, data)
+    return ray_jll.LocalMemoryBuffer(bytes, sizeof(bytes), true)
+end
```

Comment:
another thing I'm not sure about is whether the LocalMemoryBuffer created this way will prevent julia GC from removing bytes; I _think_ so but I'm not 100% sure...